crates/red_knot/src/symbols.rs

```diff
@@ -605,13 +608,21 @@ impl FlowGraph {
 
 struct SymbolTableBuilder {
     table: SymbolTable,
-    scopes: Vec<ScopeId>,
+    scopes: Vec<(ScopeId, FlowNodeId)>,
     /// the definition whose target(s) we are currently walking
     current_definition: Option<Definition>,
-    current_flow_node: FlowNodeId,
 }
 
 impl SymbolTableBuilder {
+    fn set_current_flow_node(&mut self, new_flow_node_id: FlowNodeId) {
+        let (scope_id, _) = self.scopes.pop().expect("scope stack is never empty");
+        self.scopes.push((scope_id, new_flow_node_id));
+    }
```

Comment:
The compiler might be clever enough to work this out but maybe
```suggestion
        let (_, flow_node_id) = self.scopes.last_mut().expect("scope stack is never empty");
        *flow_node_id = new_flow_node_id;
    }
```